packages/app/src/app/pages/SignIn/Onboarding.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from 'react';
+import { Element, Text, Stack } from '@codesandbox/components';
+import { useAppState, useActions } from 'app/overmind';
+import { InputText } from 'app/components/dashboard/InputText';
+import { InputSelect } from 'app/components/dashboard/InputSelect';
+import { Button } from 'app/components/dashboard/Button';
+
+const roleOptions = [
+  { value: 'frontend', label: 'Frontend developer' },
+  { value: 'backend', label: 'Backend developer' },
+  { value: 'fullstack', label: 'Fullstack developer' },
+  { value: 'designer', label: 'Designer' },
+  { value: 'product-manager', label: 'Product manager' },
+  { value: 'educator', label: 'Educator' },
+  { value: 'student', label: 'Student' },
+  { value: 'other', label: 'Other' },
+];
+
+const usageOptions = [
+  { value: 'freelance-or-client', label: 'Freelance or client work' },
+  { value: 'startup', label: 'Startup' },
+  { value: 'enterprise', label: 'Enterprise' },
+  { value: 'hobby', label: 'Hobby or fun' },
+  { value: 'education', label: 'Education' },
+  { value: 'other', label: 'Other' },
+];
+
+export const Onboarding = () => {
+  /**
+   * 🚧 Utility to debug Trial Onboarding Questions
+   */
+  const TOQ_DEBUG = window.localStorage.getItem('TOQ_DEBUG') === 'ENABLED';
+
+  // 🚧 Remove
+  let pendingUser = useAppState().pendingUser;
+
+  // 🚧 Uncomment
+  // const { pendingUser } = useAppState();
+
+  // 🚧 Remove
+  if (TOQ_DEBUG) {
+    pendingUser = {
+      avatarUrl: 'https://avatars.githubusercontent.com/u/7533849?v=4',
+      username: 'tristandubbeld',
+      name: 'Tristan Dubbeld',
+      id: 'id',
+      valid: true,
+    };
+  }
+
+  const { validateUsername, finalizeSignUp } = useActions();
+  const [newUsername, setNewUsername] = useState(pendingUser?.username || '');
+  const [newDisplayName, setNewDisplayName] = useState(pendingUser?.name || '');
+  const [loadingUsername, setLoadingUserName] = useState(false);
+  const firstName = pendingUser?.name.split(' ')[0];
+
+  return (
+    <Stack
+      direction="vertical"
+      gap={4}
+      css={{
+        width: '100%',
+        maxWidth: '370px',
+        alignSelf: 'center',
+        paddingBottom: '24px',
+      }}
+    >
+      <Stack direction="vertical" gap={6} align="center">
+        <Element
+          as="img"
+          css={{
+            width: 64,
+            height: 64,
+            borderRadius: '100%',
+          }}
+          alt={pendingUser.username}
+          src={pendingUser.avatarUrl}
+        />
+        <Text
```

Comment:
I know a lot of my comments have been about future plans, but another idea for these texts is to bring a similar API as the new typography components in prism (Heading, Paragraph, Text)

Reply:
This is a good idea, but it will take some time to properly create these. Should we pick this up at a later time?